torchgeo/datamodules/oscd.py

```diff
@@ -196,3 +197,10 @@ def test_dataloader(self) -> DataLoader[Any]:
         return DataLoader(
             self.test_dataset, batch_size=1, num_workers=self.num_workers, shuffle=False
         )
+
+    def plot(self, *args: Any, **kwargs: Any) -> plt.Figure:
+        """Run :meth:`torchgeo.datasets.OSCD.plot`.
+
+        .. versionadded:: 0.4
+        """
+        return self.test_dataset.plot(*args, **kwargs)
```

Comment:
Heads up that this won't work because the sample that comes in here will be a flattened image

Reply:
Yep, I noticed a few failures like this. Not sure what the best approach is other than to try to make things more uniform (dataset.__getitem__ should always return C x H x W, plot method should always convert to H x W x C). Prob going to take me a while to get these working uniformly. If anyone else wants to hack on this PR go ahead.

Reply:
Well the issue is that Dataset .plot() methods expects a sample as returned by __getitem__, whereas the samples that are used within the train/val methods in our trainers have been modified by the DataModule preprocessing functions

Reply:
E.g. in OSCD, `sample["image"]` as returned by `__getitem__` will be something like (2 x C x H x W) whereas in the trainer it will be converted to (2C x H x W). If we do channel-wise standardization, this will also be applied during the train/val steps.

Reply:
In that case, we should either change the dataset to return 2C x H x W, or create a new ObjectDetectionTask that can handle 2 x C x H x W samples without any changes.

Reply:
Do you mean `ChangeDetectionTask`?

Generally, what happens if the DataModule subtracts the mean and divides by the stdev?

Reply:
Yes.

Not really sure how to handle this. Kornia transforms have a reverse operation, right? Maybe we could apply that if it exists.

Reply:
Kornia has a denormalize but unfortunately it doesn't work per channel it only takes in a single value for mean and std.

Here's a torchvision denormalize I've been using lately:

```

class Denormalize(nn.Sequential):
    def __init__(
        self, mean: Sequence[float] = IMAGENET_MEAN, std: Sequence[float] = IMAGENET_STD
    ) -> None:
        self.mean = mean
        self.std = std
        self.inv_mean = torch.tensor([-x for x in mean])
        self.inv_std = torch.tensor([1 / x for x in std])
        super().__init__(
            T.Normalize(mean=[0.0, 0.0, 0.0], std=self.inv_std),
            T.Normalize(mean=self.inv_mean, std=[1.0, 1.0, 1.0]),
        )
```

Reply:
So I think we should factor out augmentations and preprocessing from the datamodules (where we can easily support denormalizing in a consistent way)